Libraries/Image/RCTImageLoader.m

```diff
@@ -417,8 +420,16 @@ - (RCTImageLoaderCancellationBlock)loadImageOrDataWithURLRequest:(NSURLRequest *
 
     // Download image
     RCTNetworkTask *task = [_bridge.networking networkTaskWithRequest:request completionBlock:^(NSURLResponse *response, NSData *data, NSError *error) {
-      if (error) {
-        completionHandler(error, nil);
+      if (error || !response || !data) {
+        NSError *someError = nil;
+        if (error) {
+          someError = error;
+        } else if (!response) {
+          someError = RCTErrorWithMessage(@"Response metadata error");
+        } else {
+          someError = RCTErrorWithMessage(@"Unknown image download error");
```

Comment:
I'm getting the "Unknown image download error" when loading images from a _localhost_ url as my app starts up:

`Object {error: "Unknown image download error", target: 235}`

An example url is: http://u59297996:p40782433@localhost:6006/pc_102_c5149e0b-d6a6-4d7c-9498-ed311552c663/d411b371795c20da92be1e84e8de540d/slice 

The images are available, and subsequent calls to load them succeed.